simplesource-command-serialization/src/main/java/io/simplesource/kafka/serialization/avro/AvroSerdes.java

```diff
@@ -107,36 +107,27 @@ private static Schema commandResponseKeySchema() {
         private static final String REASON = "reason";
         private static final String ADDITIONAL_REASONS = "additionalReasons";
         private static final String ERROR_MESSAGE = "errorMessage";
-        private static final String ERROR_CODE = "errorCode";
+        private static final String ERROR = "error";
         private static final String WRITE_SEQUENCE = "writeSequence";
         private static final String AGGREGATION = "aggregate_update";
 
         private static GenericRecord fromReason(final Schema schema, final CommandError commandError) {
             return new GenericRecordBuilder(schema)
                     .set(ERROR_MESSAGE, commandError.getMessage())
-                    .set(ERROR_CODE, commandError.getReason().name())
+                    .set(ERROR, Base64.getEncoder().encodeToString( SerializationUtils.serialize(commandError)))
```

Comment:
Inefficient in terms of CPU and space taken in the message.

The question is: How do you correctly serialize an `Exception`?

Kryo still does not provide a solution for that. See https://github.com/EsotericSoftware/kryo/issues/532
Chill doesn't help either: https://github.com/twitter/chill/issues/327

Maybe Akka can help us here. They have a `Status` type that is a kind of serializable `Try`. See https://github.com/akka/akka/blob/b006567751f1ef63588b00e20ec51c905e8562d3/akka-actor/src/main/scala/akka/actor/Actor.scala#L310

I don't know how they serialize it.